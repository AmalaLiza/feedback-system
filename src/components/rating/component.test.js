import React from "react";
import { shallow } from "enzyme";
import Rating from "./component";
import renderer from "react-test-renderer";

const props = {
	buttonText: "Button title",
	onRatingClick: (noop) => noop,
	className: "star",
	initialRating: 0,
	label: "Overall rating",
};

function setup() {
	const wrapper = shallow(<Rating {...props} />);
	return { wrapper, props };
}

describe("Rating Component Test Suite", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<Rating {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Should have a label and a Rating component", () => {
		const { wrapper } = setup();
		expect(wrapper.find("label").exists()).toBe(true);
	});
});
