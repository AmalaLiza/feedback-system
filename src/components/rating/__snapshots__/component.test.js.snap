// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Component Test Suite renders correctly 1`] = `
Array [
  <label
    className="label"
  >
    Overall rating
  </label>,
  <span
    className="star star"
    onMouseLeave={[Function]}
    style={
      Object {
        "direction": "ltr",
        "display": "inline-block",
      }
    }
  >
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
  </span>,
]
`;
